src/app/common/hooks/use-bitcoin-contracts.ts

```diff
@@ -0,0 +1,228 @@
+  const calculateFiatValue = useCalculateBitcoinFiatValue();
+  const getNativeSegwitSigner = useCurrentAccountNativeSegwitSigner();
+  const currentIndex = useCurrentAccountIndex();
+  const nativeSegwitPrivateKeychain =
+    useNativeSegwitActiveNetworkAccountPrivateKeychain()?.(currentIndex);
+
+  async function getBitcoinContractInterface(): Promise<JsDLCInterface | undefined> {
+    const bitcoinAccountDetails = getNativeSegwitSigner?.(0);
+
+    if (!nativeSegwitPrivateKeychain || !bitcoinAccountDetails) return;
+
+    const currentBitcoinNetwork = bitcoinAccountDetails.network;
+    const currentAddress = bitcoinAccountDetails.address;
+    const currentAccountIndex = bitcoinAccountDetails.addressIndex;
+
+    const currentAddressPrivateKey = deriveAddressIndexKeychainFromAccount(
+      nativeSegwitPrivateKeychain
+    )(currentAccountIndex).privateKey;
+
+    if (!currentAddressPrivateKey) return;
+
+    const blockchainAPI = whenBitcoinNetwork(currentBitcoinNetwork)({
+      mainnet: BITCOIN_API_BASE_URL_MAINNET,
+      testnet: BITCOIN_API_BASE_URL_TESTNET,
+      regtest: BITCOIN_API_BASE_URL_TESTNET,
+      signet: BITCOIN_API_BASE_URL_TESTNET,
+    });
+
+    const oracleAPI = 'https://testnet.dlc.link/oracle';
+
+    const bitcoinContractInterface = JsDLCInterface.new(
+      bytesToHex(currentAddressPrivateKey),
+      currentAddress,
+      currentBitcoinNetwork,
+      blockchainAPI,
+      oracleAPI
+    );
+
+    return bitcoinContractInterface;
+  }
+
+  function handleOffer(
+    bitcoinContractOfferJSON: string,
+    counterpartyWalletURL: string,
+    counterpartyWalletName: string,
+    counterpartyWalletIcon: string
+  ): BitcoinContractOfferDetails {
+    const bitcoinContractOffer = JSON.parse(bitcoinContractOfferJSON);
+
+    const bitcoinContractId = bitcoinContractOffer.temporaryContractId;
+    const bitcoinContractCollateralAmount =
+      bitcoinContractOffer.contractInfo.singleContractInfo.totalCollateral;
+    const bitcoinContractExpirationDate = new Date(
+      bitcoinContractOffer.cetLocktime * 1000
+    ).toLocaleDateString();
+
+    const simplifiedBitcoinContractOffer: SimplifiedBitcoinContract = {
+      bitcoinContractId,
+      bitcoinContractCollateralAmount,
+      bitcoinContractExpirationDate,
+    };
+
+    const counterpartyWalletDetails: CounterpartyWalletDetails = {
+      counterpartyWalletURL,
+      counterpartyWalletName,
+      counterpartyWalletIcon,
+    };
+
+    const bitcoinContractOfferDetails: BitcoinContractOfferDetails = {
+      simplifiedBitcoinContract: simplifiedBitcoinContractOffer,
+      counterpartyWalletDetails,
+    };
+
+    return bitcoinContractOfferDetails;
+  }
+
+  async function handleAccept(
+    bitcoinContractJSON: string,
+    counterpartyWalletDetails: CounterpartyWalletDetails
+  ) {
+    const bitcoinContractInterface = await getBitcoinContractInterface();
+
+    if (!bitcoinContractInterface) return;
```

Comment:
shouldn't the logic here be like in `catch` block? if there is no interface => lead to error page